src/versionUtils.ts

```diff
@@ -0,0 +1,159 @@
+	}
+}
+
+export async function getLatestCamelKVersion(): Promise<Errorable<string>> {
+	const latestURL = 'https://api.github.com/repos/apache/camel-k/releases/latest';
+	const downloadResult = await downloader.toTempFile(latestURL);
+	if (failed(downloadResult)) {
+		return { succeeded: false, error: [`Failed to establish Apache Camel K stable version: ${downloadResult.error[0]}`] };
+	}
+	const rawtext = fs.readFileSync(downloadResult.result, 'utf-8');
+	let latestJSON = JSON.parse(rawtext);
+	let tagName = latestJSON.tag_name;
+	if (tagName) {
+		return { succeeded: true, result: tagName };
+	}
+	return { succeeded: false, error: [`Failed to retrieve latest Apache Camel K version tag from : ${latestURL}`] };
+}
+
+function checkKamelCLIVersion() : Promise<string> {
+	return new Promise<string>( async (resolve, reject) => {
+		let kamelLocal = kamelCli.create();
+		await kamelLocal.invoke('version')
+			.then( (rtnValue) => {
+				const strArray = rtnValue.split(' ');
+				const version = strArray[strArray.length - 1].trim();
+				console.log(`Apache Camel K CLI (kamel) version returned: ${version}`);
+				resolve(version);
+				return;
+			}).catch (kamelUnavailableRejection(reject));
+	});
+}
+
+export async function handleChangeRuntimeConfiguration() {
+	let runtimeSetting = config.getKamelRuntimeVersionConfig();
+	let newUpgradeSetting = config.getKamelAutoupgradeConfig();
+	if (newUpgradeSetting === true) {
+		if (runtimeSetting && version.toLowerCase() !== runtimeSetting.toLowerCase()) {
+			extension.shareMessageInMainOutputChannel(`Auto-upgrade setting enabled. Updating to default version ${version} of Apache Camel K CLI`);
+			await config.setKamelRuntimeVersionConfig(version).then ( () => {
+				const msg = `Setting for Apache Camel K runtime version has changed to default ${version}. Please restart the workspace to refresh the Camel K cli.`;
+				extension.shareMessageInMainOutputChannel(msg);
+				vscode.window.showWarningMessage(msg);
+				extension.setRuntimeVersionSetting(version);	
+			});
+		}
+	} else if (extension.runtimeVersionSetting) {
+		if (runtimeSetting && runtimeSetting.trim().length > 0 
+			&& extension.runtimeVersionSetting.toLowerCase() !== runtimeSetting.toLowerCase()) {
+			const msg = `Setting for Apache Camel K runtime version has changed to ${runtimeSetting}. Please restart the workspace to refresh the Camel K cli.`;
+			extension.shareMessageInMainOutputChannel(msg);
+			vscode.window.showWarningMessage(msg);
+			extension.setRuntimeVersionSetting(runtimeSetting);	
+		}
+	} else if (runtimeSetting && runtimeSetting.trim().length > 0 && version.toLowerCase() !== runtimeSetting.toLowerCase()) {
+		const msg = `Setting for Apache Camel K runtime version has changed to ${runtimeSetting}. Please restart the workspace to refresh the Camel K cli.`;
+		extension.shareMessageInMainOutputChannel(msg);
+		vscode.window.showWarningMessage(msg);
+		extension.setRuntimeVersionSetting(runtimeSetting);	
```

Comment:
I have hard time to understand this piece of code.
Can you provide more specific function for if statement or if-block please?

Also, i the think the 3 lines can be factorized		```extension.shareMessageInMainOutputChannel(msg);
		vscode.window.showWarningMessage(msg);
		extension.setRuntimeVersionSetting(runtimeSetting);```

Reply:
Will see what I can do

Reply:
Added docs and refactored with suggestion

Reply:
The comments are helping to understand the code even if having refactored code with clear method names would have been better as it would ensure a better maintainability.